pkg/api/entries.go

```diff
@@ -272,54 +272,55 @@ func getEntryURL(locationURL url.URL, uuid string) strfmt.URI {
 
 // GetLogEntryByUUIDHandler gets log entry and inclusion proof for specified UUID aka merkle leaf hash
 func GetLogEntryByUUIDHandler(params entries.GetLogEntryByUUIDParams) middleware.Responder {
-	ctx := params.HTTPRequest.Context()
-
 	uuid, err := sharding.GetUUIDFromIDString(params.EntryUUID)
 	if err != nil {
 		return handleRekorAPIError(params, http.StatusBadRequest, err, "")
 	}
-	var tid int64
 	tidString, err := sharding.GetTreeIDFromIDString(params.EntryUUID)
-	if err != nil {
-		// If EntryID is plain UUID, assume no sharding and use ActiveTreeID. The ActiveTreeID
-		// will == the tlog_id if a tlog_id is passed in at server startup.
-		if err.Error() == "cannot get treeID from plain UUID" {
-			tid = api.logRanges.ActiveTreeID()
-		} else {
-			return handleRekorAPIError(params, http.StatusBadRequest, err, "")
-		}
-	} else {
-		tid, err = strconv.ParseInt(tidString, 16, 64)
+
+	switch err {
+	// No error means we can route to the correct treeID from the entryID
+	case nil:
+		tid, err := strconv.ParseInt(tidString, 16, 64)
 		if err != nil {
 			return handleRekorAPIError(params, http.StatusBadRequest, err, "")
 		}
-	}
-	hashValue, _ := hex.DecodeString(uuid)
-
-	tc := NewTrillianClientFromTreeID(params.HTTPRequest.Context(), tid)
-	log.RequestIDLogger(params.HTTPRequest).Debugf("Retrieving UUID %v from TreeID %v", uuid, tid)
+		resp, _ := AttemptRetrieveUUID(params, uuid, tid)
+		return resp
 
-	resp := tc.getLeafAndProofByHash(hashValue)
-	switch resp.status {
-	case codes.OK:
-	case codes.NotFound:
-		return handleRekorAPIError(params, http.StatusNotFound, fmt.Errorf("grpc error: %w", resp.err), "")
+	// If there is an error, see if it is because the EntryID is a UUID and if so,
+	// handle it
 	default:
```

Comment:
The switch statement within the switch statement is a little hard to read. How about a structure like this:

```golang
tidString, err := sharding.GetTreeIDFromIDString(params.EntryUUID)
if err == nil {
   // Get the UUID from the specified TreeID
}
// This should get the active tree ID and the inactive IDs
trees, err := GetTrees()
if err != nil {
  return handleRekorAPIError(params, http.StatusBadRequest, err, "")
}
for _, t := range trees {
  logEntry, err := RetrieveUUID(params, uuid, t)
  if err != nil {
    continue
  }
  return entries.NewGetLogEntryByUUIDOK().WithPayload(logEntry)
}
// If we've reached this point, none of the shards have worked out, so return an error
return handleRekorAPIError(params, http.StatusBadRequest, fmt.Errorf("unable to find UUID"), "")
```

Reply:
I changed it to be more like this, let me know what you think about the readability.